rotkehlchen/chain/ethereum/modules/makerdao/sai/decoder.py

```diff
@@ -0,0 +1,535 @@
+                amount = asset_normalized_value(amount=amount_raw, asset=self.weth)
+
+                event = HistoryBaseEntry(
+                    event_identifier=transaction.tx_hash,
+                    timestamp=ts_sec_to_ms(transaction.timestamp),
+                    sequence_index=self.base_tools.get_sequence_index(tx_log),
+                    location=Location.BLOCKCHAIN,
+                    asset=self.peth,
+                    balance=Balance(amount=amount),
+                    event_type=HistoryEventType.SPEND,
+                    event_subtype=HistoryEventSubType.SPEND,
+                    location_label=liquidator,
+                    notes=f'Liquidate {amount} PETH for CDP {cdp_id}',
+                    counterparty=CPT_SAI,
+                )
+                return event, []
+
+        return None, []
+
+    def _decode_deposit_eth_event(
+            self,
+            tx_log: EvmTxReceiptLog,  # pylint: disable=unused-argument
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],
+            action_items: Optional[list[ActionItem]],  # pylint: disable=unused-argument
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        """
+        This method decodes a deposit event of ETH into a CDP.
+
+        An example of such transaction is:
+        https://etherscan.io/tx/0x5a7849ab4b7f7de2b005deddef24a094387c248c3bcb06066109bd7852c1d8af
+        """
+        # check if ETH is sent to sai proxy contract or a proxy and mark it
+        # as a deposit since the proxy handles the conversion to WETH
+        # and transfer to the sai tub contract
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.counterparty == MAKERDAO_SAI_PROXY_CONTRACT and
+                event.asset == self.eth
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Supply {event.balance.amount} {self.eth.symbol} to SAI vault'  # noqa: E501
+                event.counterparty = CPT_SAI
+                return None, []
+
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == self.eth
+            ):
+                # check for PETH mint event to match the deposit
+                for log in all_logs:
+                    if (
+                        log.topics[0] == PETH_MINT_EVENT_TOPIC and
+                        log.address == POOLED_ETHER_ADDRESS and
+                        hex_or_bytes_to_int(tx_log.topics[2]) == hex_or_bytes_to_int(log.data[:32])
```

Comment:
Need explanation of those two ints as a comment above. Explain what each int is.
```suggestion
                       #explain
                        hex_or_bytes_to_int(tx_log.topics[2]) == hex_or_bytes_to_int(log.data[:32])
```